tests/test_pandas_parser.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+from io import StringIO
+
+import pyarrow as pa
+import pandas as pd
+import numpy as np
+from pandas.testing import assert_frame_equal, assert_series_equal
+from arrow_pd_parser.parse import pd_read_csv, cast_pandas_table_to_schema
+from arrow_pd_parser.parse.pandas_parser import _infer_bool_type, convert_to_bool_series
+
+from jsonschema.exceptions import ValidationError
+
+from mojap_metadata import Metadata
+
+
+def test_file_reader_returns_df():
+    csv_meta = {
+        "columns": [
+            {"name": "test", "type_category": "string"},
+            {"name": "a_column", "type_category": "string"},
+        ]
+    }
+
+    df = pd_read_csv("tests/data/example_data.csv", metadata=csv_meta)
+    assert isinstance(df, pd.DataFrame)
+    assert [str(df[c].dtype) for c in df.columns] == ["string", "string"]
+
+    # df = pa_read_json("tests/data/example_data.jsonl")
+    # assert isinstance(df, pd.DataFrame)
+
+
+def test_file_reader_works_with_both_meta_types():
+    csv_meta = {
+        "columns": [
+            {"name": "test", "type_category": "string"},
+            {"name": "a_column", "type_category": "string"},
+        ]
+    }
+    df_csv1 = pd_read_csv("tests/data/example_data.csv", csv_meta)
+    csv_meta = Metadata.from_dict(csv_meta)
+    df_csv2 = pd_read_csv("tests/data/example_data.csv", csv_meta)
+    assert_frame_equal(df_csv1, df_csv2)
+    with pytest.raises(KeyError):
+        pd_read_csv("tests/data/example_data.jsonl", metadata={})
+    with pytest.raises(ValidationError):
+        m = {"columns": [{"name": "broken", "type": "not-a-type"}]}
+        pd_read_csv("tests/data/example_data.jsonl", metadata=m)
+
+
+def test_basic_end_to_end():
+    data = """
```

Comment:
Do we not want to add float type to `all_types.csv` - it doesn't seem to cover all types!

Reply:
Yeah it's because the all types isn't actually used that much but maybe worth doing anyway for consistency.